core.js

```diff
@@ -686,6 +687,41 @@ function tickAnimationFrame() {
 
     tickAnimationFrame.window = null;
   }
+
+  if (timeouts.length > 0) {
+    const dateNow = Date.now();
+    const removedTimeoutIndexes = {};
```

Comment:
We can try to avoid object creations and function creations in the render loop.

e.g., also `timeouts.filter((t, i) => !removedTimeoutIndexes[i]);`, should pre-defined the function or use a reusable array that's cleared each run.

Reply:
Yeah, good call. If there's one place we should eliminate allocs it's this code path.

Reply:
Added in https://github.com/webmixedreality/exokit/pull/224/commits/639e05cbaccc17bac1e9f3a7353948918e868ca2, which should also be a general rAF optimization since we [used to create garbage in the processing](https://github.com/webmixedreality/exokit/blob/b89309146d6520d0e60a6438687578bcc8cf05ac/core.js#L658).